pkg/cloudprovider/vsphere/route/routeprovider.go

```diff
@@ -0,0 +1,182 @@
+	for _, item := range staticRoutes.Results {
+		nodeName := ""
+		tagsField, _ := item.Field("tags")
+		tags := (tagsField).(*data.ListValue).List()
+		for _, tItem := range tags {
+			scope, _ := (tItem).(*data.StructValue).Field("scope")
+			if (scope).(*data.StringValue).Value() == config.NodeNameTagScope {
+				tag, _ := (tItem).(*data.StructValue).Field("tag")
+				nodeName = (tag).(*data.StringValue).Value()
+				break
+			}
+		}
+		routeId, _ := item.Field("id")
+		network, _ := item.Field("network")
+		route := &cloudprovider.Route{
+			Name:            (routeId).(*data.StringValue).Value(),
+			TargetNode:      types.NodeName(nodeName),
+			DestinationCIDR: (network).(*data.StringValue).Value(),
+		}
+		routes = append(routes, route)
+	}
+	return routes, nil
+}
+
+// CreateRoute creates a static route on NSXT for a Node
+func (p *RouteProvider) CreateRoute(ctx context.Context, clusterName string, nameHint string, route *cloudprovider.Route) error {
+	nodeName := string(route.TargetNode)
+	klog.V(6).Infof("Creating static route for node %s", nodeName)
+
+	nodeIp, err := p.getNodeIPAddress(nodeName)
+	if err != nil {
+		klog.Errorf("Getting node %s IP address failed: %v", nodeName, err)
+		return err
+	}
+
+	var tags []model.Tag
+	clusterNameScope := config.ClusterNameTagScope
+	nodeNameScope := config.NodeNameTagScope
+	tags = append(tags, model.Tag{Scope: &clusterNameScope, Tag: &clusterName})
+	tags = append(tags, model.Tag{Scope: &nodeNameScope, Tag: &nodeName})
+	var nexthops []model.RouterNexthop
+	nexthops = append(nexthops, model.RouterNexthop{IpAddress: &nodeIp})
+	staticRoute := model.StaticRoutes{
+		Network:  route.DestinationCIDR,
+		NextHops: nexthops,
+		Tags:     tags,
+	}
+	routerId := p.getRouterId(p.routerPath)
+	routeId := nameHint + "_" + route.DestinationCIDR
+	routeId = strings.ReplaceAll(routeId, "/", "_")
+	err = p.broker.tier1StaticRoutesClient.Patch(routerId, routeId, staticRoute)
+	if err != nil {
+		klog.Errorf("Creating static route %s for node %s failed: %s", routeId, nodeName, err)
+		return err
+	}
+
+	// Get realized state
+	return p.checkStaticRouteRealizedState(p.routerPath, routeId)
+}
+
+// DeleteRoute deletes Node's static route on NSXT
+func (p *RouteProvider) DeleteRoute(ctx context.Context, clusterName string, route *cloudprovider.Route) error {
+	klog.V(6).Infof("Deleting static route %s on router %s in cluster %s",
+		route.Name, p.routerPath, clusterName)
+	routerId := p.getRouterId(p.routerPath)
+	err := p.broker.tier1StaticRoutesClient.Delete(routerId, route.Name)
+	if err != nil {
+		klog.Errorf("Deleting static route %s failed: %s", route.Name, err)
+		return err
+	}
+	return nil
+}
+
+// getNodeIPAddress gets node IP address
+// The order is to choose node internal IP first, then external IP.
+func (p *RouteProvider) getNodeIPAddress(nodeName string) (string, error) {
+	node, err := p.kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Comment:
Can we pass in a node informer cache to `RouterProvider` and have this call get from cache? An API call for every route reconcile will get expensive quickly.

Reply:
You can pass a shared informer through `Initialize`. Or alternatively we can start the cache informer there and worry about using shared informers in a follow-up PR.

Reply:
OK, let me check. 
Actually we only need get node when route controller calls CreateRoute(), which happens when there is a new node added. In every route reconcile basically it only calls ListRoutes()

Reply:
> Actually we only need get node when route controller calls CreateRoute(), which happens when there is a new node added. In every route reconcile basically it only calls ListRoutes(

CreateRoute would also be called if ListRoutes doesn't return the route of a node, which can happen if `CreateRoute` errors repeatedly. I agree though that `CreateRoute` is generally not called as frequently as `ListRoutes`. Maybe this can be done in a follow-up PR. 

I think it's worthwhile to consider rate limiting or cachingi `ListRoutes` results though, as that can be an expensive API call, especially when done frequently by the routes controller.

Reply:
Sure, make sense to me. Based on my test, router controller calls ListRoutes every 10 seconds. We can have a cache in cloud provider to avoid frequent NSXAPI call.

Reply:
sgtm, I'd be okay to add caching support in a follow-up, but should be done prior to an official release I think.

Reply:
Just saw we already have an informer for node manager, maybe I can add logic in `nodeAdded` and `nodeDeleted` to call some route interfaces to cache node info